coffee-module/coffee-module-mp/coffee-module-mp-opentracing/src/main/java/hu/icellmobilsoft/coffee/module/mp/opentracing/extension/BaseOpenTraceInterceptor.java

```diff
@@ -0,0 +1,166 @@
+ * based on {@link io.opentracing.contrib.interceptors.OpenTracingInterceptor}
+ * 
+ * @author czenczl
+ * @since 1.7.0
+ */
+public abstract class BaseOpenTraceInterceptor {
+
+    /**
+     * handle snap join to parent based on {@link io.opentracing.contrib.interceptors.OpenTracingInterceptor#SPAN_CONTEXT}
+     */
+    public static final String SPAN_CONTEXT = OpenTracingInterceptor.SPAN_CONTEXT;
+
+    private static final Logger LOGGER = Logger.getLogger(BaseOpenTraceInterceptor.class);
+
+    @Inject
+    private OpenTraceResolver openTraceResolver;
+
+    /**
+     * Get the underlying {@link Tracer} implementation
+     * 
+     * @return {@link Tracer}
+     */
+    protected Tracer getTracer() {
+        return openTraceResolver.resolveTracer();
+    }
+
+    /**
+     * Handle span creation based on interception context data
+     * 
+     * @param ctx
+     *            {@link InvocationContext}
+     * @param spanBuilder
+     *            {@link SpanBuilder}
+     * @return {@link InvocationContext#proceed()}
+     * @throws Exception
+     *             if error
+     */
+    protected Object handleSpan(InvocationContext ctx, Tracer.SpanBuilder spanBuilder) throws Exception {
+        Tracer tracer = openTraceResolver.resolveTracer();
+        int contextParameterIndex = -1;
+        for (int i = 0; i < ctx.getParameters().length; i++) {
+            Object parameter = ctx.getParameters()[i];
+            if (parameter instanceof SpanContext) {
+                LOGGER.debug("Found parameter as span context. Using it as the parent of this new span");
+                spanBuilder.asChildOf((SpanContext) parameter);
+                contextParameterIndex = i;
+                break;
+            }
+
+            if (parameter instanceof Span) {
+                LOGGER.debug("Found parameter as span. Using it as the parent of this new span");
+                spanBuilder.asChildOf((Span) parameter);
+                contextParameterIndex = i;
+                break;
+            }
+        }
+
+        if (contextParameterIndex < 0) {
+            LOGGER.debug("No parent found. Trying to get span context from context data");
+            Object ctxParentSpan = ctx.getContextData().get(SPAN_CONTEXT);
+            if (ctxParentSpan instanceof SpanContext) {
+                LOGGER.debug("Found span context from context data.");
+                SpanContext parentSpan = (SpanContext) ctxParentSpan;
+                spanBuilder.asChildOf(parentSpan);
+            }
+        }
+
+        Span span = spanBuilder.start();
+        Scope scope = tracer.activateSpan(span);
+        try {
+            LOGGER.debug("Adding span context into the invocation context.");
+            ctx.getContextData().put(SPAN_CONTEXT, span.context());
+
+            if (contextParameterIndex >= 0) {
+                LOGGER.debug("Overriding the original span context with our new context.");
+                for (int i = 0; i < ctx.getParameters().length; i++) {
```

Comment:
kell ez a for? minden i-re fixen a contextParameterIndex indexen lévő érték állítódik?